action.yml

```diff
@@ -7,36 +7,52 @@ inputs:
     default: 1.10.2
 runs:
   using: "composite"
+
   steps:
     - name: Set up cache paths
       id: cache-paths
       shell: bash
       run: |
-        echo "setupsbt_path=$HOME/work/_setupsbt" >> $GITHUB_OUTPUT
+        echo "sbt_toolpath=$RUNNER_TOOL_CACHE/sbt/${{ inputs.sbt-runner-version }}" >> "$GITHUB_OUTPUT"
+        echo "sbt_downloadpath=$RUNNER_TEMP/_sbt" >> "$GITHUB_OUTPUT"
+
+    - name: Check Tool Cache
+      id: cache-tool-dir
+      shell: bash
+      run: |
+        mkdir -p "${{ steps.cache-paths.outputs.sbt_toolpath }}"
+        if [ -f "${{ steps.cache-paths.outputs.sbt_toolpath }}/sbt/bin/sbt" ]; then
+          echo "cache-hit=true" >> "$GITHUB_OUTPUT"
+        else
+          echo "cache-hit=false" >> "$GITHUB_OUTPUT"
+        fi
 
     - name: Cache sbt distribution
       id: cache-dir
+      if: steps.cache-tool-dir.outputs.cache-hit != 'true'
       uses: actions/cache@v4
       with:
-        path: ${{ steps.cache-paths.outputs.setupsbt_path }}
-        key: ${{ runner.os }}-sbt-${{ inputs.sbt-runner-version }}-1.1.1
+        path: ${{ steps.cache-paths.outputs.sbt_toolpath }}
```

Comment:
Note: because the `$RUNNER_TOOL_CACHE` path can be different between self-hosted and hosted runners, they may not be able to share this cache (actions/cache uses the paths in the cache key), but have separate caches.  In practice I don't think this matters because of how fast the curl is, and how few different versions of sbt will need to be cached (its not like the cache between the `build and test` and `publish` on the typical `sbt-typelevel` project where its important to not rebuild the project).